### PR TITLE
XML output for DeckLink capture

### LINKED ISSUE #802: support xml output for a decklink input
example:

```
<?xml version="1.0" encoding="UTF-8"?>
<dvrescue xmlns="https://mediaarea.net/dvrescue" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://mediaarea.net/dvrescue https://mediaarea.net/dvrescue/dvrescue.xsd" version="1.2.1">
	<creator>
		<program>dvrescue</program>
		<version>22.12.20231208</version>
		<library version="23.11">MediaInfoLib</library>
	</creator>
	<media ref="out.mkv" fromCapture="device://84:3c0a102a:00000000" deviceName="UltraStudio HD Mini" format="Matroska" size="9120000">
        <source video="sdi" audio="embedded" timecode="vitc"/>
		<stop method='user' extra='SIGINT'/>
		<frames count="76" pts="00:00:00.000000" end_pts="00:00:02.535866" size="720x486" video_rate="30000/1001" chroma_subsampling="4:2:2" audio_rate="48000" channels="2">
			<frame n="0" pos="0" pts="00:00:00.000000" tc="01:16:11:01" seqn="F"/>
			<frame n="75" pos="9000000" pts="00:00:02.502500" tc="01:16:13:16" seqn="F"/>
		</frames>
	</media>
</dvrescue>
```

This is the same as for dv, but removes `abst` and adds a few attributes (the `source` ones and `deviceName`. I'd like the same approach with `tc_n`.

Eventually, I'd like to add in some libavfilter generated metadata, such as the `TAG:lavfi.signalstats.SATMAX`.

### PR DESCRIPTION
**Pull request by g-maxime**
Fixes #802

### CODE REVIEW

**Comment by dericed** [comment]
:D

**Comment by dericed** [comment]
with snapshot:
```
$ dvrescue device://84:3c0a102a:00000000 --control "/dev/tty.RS422 Deck Control" -x test.dvrescue.xml -m test.mkv --csv --verbosity 9
Capture frame 1, press ctrl+c to stop.Segmentation fault: 11
```

**Comment by g-maxime** [comment]
[Snapshot](https://mediaarea.net/download/snapshots/binary/dvrescue/20240218/dvrescue_CLI_22.12.20240218_Mac.dmg)

**Comment by dericed** [comment]
On my M1 laptop:

```
$ dvrescue device://84:160a010c:00000000 --control "/dev/tty.RS422 Deck Control" --capture -x test81.xml -m test81.mkv
2024-02-27 11:55:01.009 dvrescue[3808:38443] WARNING: AVCaptureDeviceTypeExternal is deprecated for Continuity Cameras. Please use AVCaptureDeviceTypeContinuityCamera and add NSCameraUseContinuityCameraDeviceType to your Info.plist.
Capture frame 1, press c
[test81.mkv.zip](https://github.com/mipops/dvrescue/files/14423248/test81.mkv.zip)
trl+c to stop.Segmentation fault: 11
```

xml is empty and the mkv is attached

**Comment by g-maxime** [comment]
Snapshot: https://mediaarea.net/download/snapshots/binary/dvrescue/20240227/dvrescue_CLI_22.12.20240227_Mac.dmg

**Comment by dericed** [comment]
Ok, retesting. 

```
$ dvrescue device://84:160a010c:00000000 --control "/dev/tty.RS422 Deck Control" --capture -x test85.xml -m test85.mkv
2024-02-27 15:01:17.564 dvrescue[18700:168048] WARNING: AVCaptureDeviceTypeExternal is deprecated for Continuity Cameras. Please use AVCaptureDeviceTypeContinuityCamera and add NSCameraUseContinuityCameraDeviceType to your Info.plist.
Capture frame 14, press ctrl+c to stop.drice:dvrescue drice$ 
```

basically stops on its own right away and gives 3 files [test85.zip](https://github.com/mipops/dvrescue/files/14425061/test85.zip)

I get a little xml though 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<dvrescue xmlns="https://mediaarea.net/dvrescue" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://mediaarea.net/dvrescue https://mediaarea.net/dvrescue/dvrescue.xsd" version="1.2.1">
	<creator>
		<program>dvrescue</program>
		<version>22.12.20240227</version>
		<library version="24.01">MediaInfoLib</library>
	</creator>
	<media ref="test85.mkv" fromCapture="device://84:160a010c:00000000" format="Matroska" size="13244214">
		<source video="sdi" audio="embedded" timecode="vitc">
		<frames count="14" pts="00:00:00.000000" end_pts="00:00:00.467133" size="720x486" video_rate="30000/1001" audio_rate="48000" channels="2">
			<frame n="0" pts="00:00:00.000000"/>
			<frame n="1" pts="00:00:00.033366"/>
			<frame n="2" pts="00:00:00.066733"/>
			<frame n="3" pts="00:00:00.100100"/>
			<frame n="4" pts="00:00:00.133466"/>
			<frame n="5" pts="00:00:00.166833"/>
			<frame n="6" pts="00:00:00.200200"/>
			<frame n="7" pts="00:00:00.233566"/>
			<frame n="8" pts="00:00:00.266933"/>
			<frame n="9" pts="00:00:00.300300"/>
			<frame n="10" pts="00:00:00.333666"/>
			<frame n="11" pts="00:00:00.367033"/>
			<frame n="12" pts="00:00:00.400400"/>
			<frame n="13" pts="00:00:00.433766"/>
		</frames>
	</media>
</dvrescue>
```

I then reran things when the deck was already playing and that time the capture kept running.

```
$ dvrescue device://84:160a010c:00000000 --control "/dev/tty.RS422 Deck Control" --capture -x test86.xml -m test86.mkv
2024-02-27 15:03:16.238 dvrescue[19136:172158] WARNING: AVCaptureDeviceTypeExternal is deprecated for Continuity Cameras. Please use AVCaptureDeviceTypeContinuityCamera and add NSCameraUseContinuityCameraDeviceType to your Info.plist.
Capture frame 167, press ctrl+c to stop.^Cdrice:dvrescue drice$ 
```

3 files out again
[test86.zip](https://github.com/mipops/dvrescue/files/14425072/test86.zip)

```xml
<?xml version="1.0" encoding="UTF-8"?>
<dvrescue xmlns="https://mediaarea.net/dvrescue" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://mediaarea.net/dvrescue https://mediaarea.net/dvrescue/dvrescue.xsd" version="1.2.1">
	<creator>
		<program>dvrescue</program>
		<version>22.12.20240227</version>
		<library version="24.01">MediaInfoLib</library>
	</creator>
	<media ref="test86.mkv" fromCapture="device://84:160a010c:00000000" format="Matroska" size="157985064">
		<stop method='user' extra='SIGINT'/>
		<source video="sdi" audio="embedded" timecode="vitc">
		<frames count="167" pts="00:00:00.000000" end_pts="00:00:05.572233" size="720x486" video_rate="30000/1001" audio_rate="48000" channels="2">
			<frame n="0" pts="00:00:00.000000" tc="00:03:32;08"/>
			<frame n="1" pts="00:00:00.033366" tc="00:03:32;09"/>
			<frame n="2" pts="00:00:00.066733" tc="00:03:32;10"/>

<!-- removed -->

			<frame n="165" pts="00:00:05.505500" tc="00:03:37;25"/>
			<frame n="166" pts="00:00:05.538866" tc="00:03:37;25"/>
		</frames>
	</media>
</dvrescue>
```

**Comment by dericed** [comment]
ah, some xml errors:

the `<source>` tag opens but never closes

`<source video="sdi" audio="embedded" timecode="vitc">`

If I change that element to `<source video="sdi" audio="embedded" timecode="vitc"/>` then it's valid xml but not valid to schema:

```
$ xmllint --noout --schema tools/dvrescue.xsd test86.xml
test86.xml:10: element source: Schemas validity error : Element '{https://mediaarea.net/dvrescue}source': This element is not expected. Expected is ( {https://mediaarea.net/dvrescue}frames ).
test86.xml fails to validate
```

File: Source/Common/Output_Xml.cpp [hunk 1]
```diff
@@ -202,6 +303,124 @@ return_value Output_Xml(ostream& Out, std::vector<file*>& PerFile, bitset<Option
...
+            }
+
+            auto TimeCode = decklink_timecodeformat_to_string(DeckLinkTimecodeFormat);
+            if (!TimeCode.empty())
+            {
+                Text += " timecode=\"";
+                Text += TimeCode;
+                Text += '\"';
+            }
+            Text += ">\n";
```
Comment (dericed) [inline Source/Common/Output_Xml.cpp:335]: The source node opens here, but never closes. Should this be `Text += "/>\n";`?

**Comment by dericed** [comment]
@g-maxime can you also add the `pts` attribute?

**Pull request merged**